### PR TITLE
Move JsInterop files to javascript folder

### PR DESCRIPTION
**Pull request by jrdouceur**
wip #30

### CODE REVIEW

**Review by rido-min — CHANGES_REQUESTED** [review]
We are not going to produce NPM packages (for now)
Move the JSInterop project to the samples folder with a different name
We dont need any additional scripts/pipelines 

Where are the .d.ts files?

**Comment by jrdouceur** [comment]
> We are not going to produce NPM packages (for now)

Agreed.

> Move the JSInterop project to the samples folder with a different name

I need to think of a different name.  At present, 'samples' is really a collection of tutorials; perhaps I should rename it to 'tutorials' and leave the name 'samples' for a less structured collection of coding examples?

Are you suggesting moving the entire 'javascript' folder to under 'samples'?  Or are you proposing a split?  If so, what goes where?

> We dont need any additional scripts/pipelines

Are we not going to make sure this builds?  Even if it's just a sample?

> Where are the .d.ts files?

They are currently:

* javascript\DtdlErr.d.ts
* javascript\index.d.ts
* generated\DtdlOm.d.ts

Where do you think they should go?

**Comment by rido-min** [comment]
Tutorials/Samples doesnt matter, I will keep those as samples that will be built with the samples.sln

I'll keep the .d.ts files in the javascript folder.

**Comment by jrdouceur** [comment]
> Tutorials/Samples doesnt matter, I will keep those as samples that will be built with the samples.sln

The current samples.sln is a generated file.  I'm going to start a separate PR that moves the extant samples to 'tutorials'.  Then, I'll come back to this PR and use it as a first entry in a new 'samples' folder and solution.

**Comment by jrdouceur** [comment]
> > Tutorials/Samples doesnt matter, I will keep those as samples that will be built with the samples.sln
> 
> The current samples.sln is a generated file. I'm going to start a separate PR that moves the extant samples to 'tutorials'. Then, I'll come back to this PR and use it as a first entry in a new 'samples' folder and solution.

Got a PR out for the sample-to-tutorial change:  https://github.com/digitaltwinconsortium/DTDLParser/pull/24

File: samples/JsInterop/JsInterop.csproj [hunk 1]
```diff
@@ -2,15 +2,24 @@
...
+    <RunAOTCompilation>false</RunAOTCompilation>
+    <PublishTrimmed>true</PublishTrimmed>
+    <InvariantGlobalization>true</InvariantGlobalization>
+    <EmccInitialHeapSize>16mb</EmccInitialHeapSize>
+    <DebuggerSupport>false</DebuggerSupport>
+    <UseSystemResourceKeys>true</UseSystemResourceKeys>
+    <EventSourceSupport>false</EventSourceSupport>
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TF_BUILD)' == 'true'">
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:17]: do we need this TF_Build?

File: samples/JsInterop/JsInterop.csproj [hunk 2]
```diff
@@ -2,15 +2,24 @@
 
   <PropertyGroup>
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:3]: I'd suggest to rename the project to DtdlParserJSInteropSample

File: samples/JsInterop/JsInterop.csproj [hunk 3]
```diff
@@ -2,15 +2,24 @@
...
   </PropertyGroup>
 
   <PropertyGroup Condition="'$(TF_BUILD)' == 'true'">
     <ContinuousIntegrationBuild>true</ContinuousIntegrationBuild>
   </PropertyGroup>
 
   <PropertyGroup>
-    <OutputType>Library</OutputType>
+    <OutputType>Exe</OutputType>
     <PackageLicenseExpression>MIT</PackageLicenseExpression>
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:23]: if we are not packaging we dont need this

File: samples/JsInterop/JsInterop.csproj [hunk 4]
```diff
@@ -2,15 +2,24 @@
...
     <Copyright>© Digital Twin Consortium and contributors. All rights reserved.</Copyright>
     <PackageId>DTDLParserInterop</PackageId>
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:25]: no packagaing, this is not needed

File: samples/JsInterop/JsInterop.csproj [hunk 5]
```diff
@@ -36,6 +45,13 @@
     <DocumentationFile>$(OutputPath)DTDLParserInterop.xml</DocumentationFile>
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:45]: I dont think we neee code comments for a sample

File: samples/JsInterop/JsInterop.csproj [hunk 6]
```diff
@@ -36,6 +45,13 @@
...
 
+  <ItemGroup>
+    <WasmExtraFilesToDeploy Include="..\..\javascript\DtdlErr.d.ts" />
+    <WasmExtraFilesToDeploy Include="..\..\javascript\generated\DtdlOm.d.ts" />
+    <WasmExtraFilesToDeploy Include="index.d.ts" />
+    <WasmExtraFilesToDeploy Include="package.json" />
+  </ItemGroup>
+
   <ItemGroup>
     <None Remove="stylecop.json" />
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:56]: I would not add stylecop (neither GlobalSuppressions) in this sample

File: samples/JsInterop/JsInterop.csproj [hunk 7]
```diff
@@ -36,6 +45,13 @@
     <DocumentationFile>$(OutputPath)DTDLParserInterop.xml</DocumentationFile>
   </PropertyGroup>
 
+  <ItemGroup>
+    <WasmExtraFilesToDeploy Include="..\..\javascript\DtdlErr.d.ts" />
+    <WasmExtraFilesToDeploy Include="..\..\javascript\generated\DtdlOm.d.ts" />
+    <WasmExtraFilesToDeploy Include="index.d.ts" />
+    <WasmExtraFilesToDeploy Include="package.json" />
```
Comment (rido-min) [inline samples/JsInterop/JsInterop.csproj:52]: we dont need package.json

**Review by rido-min — COMMENTED** [review]
some nits,

basically remove package related items.. As this might require additional edits/updates I can take it from here, once it's merged.

File: samples/JsInterop/index.js [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+    let objectModelJson
+    try {
+        objectModelJson = await dotnetExports.ModelParserInterop.ParseToJsonAsync(Array.from(jsonTexts))
+    } catch (modelParsingError) {
+        try {
+            /**
+             * Error info was marshalled accross the interop boundary using the error message.
+             * Parse the error info JSON into a more usable form in JS.
+             */
+            modelParsingError.exceptionInfo = JSON.parse(error.message)
```
Comment (vishnureddy17) [inline samples/JsInterop/index.js:27]: Caught by @jrdouceur in the previous PR
```suggestion
            modelParsingError.exceptionInfo = JSON.parse(modelParsingError.message)
```

**Review by vishnureddy17 — COMMENTED** [review]
LGTM besides the one error. I'm good with @rido-min's feedback as well.

File: samples/JsInterop/JsInterop.csproj [hunk 9]
```diff
@@ -36,6 +45,13 @@
     <DocumentationFile>$(OutputPath)DTDLParserInterop.xml</DocumentationFile>
   </PropertyGroup>
 
+  <ItemGroup>
+    <WasmExtraFilesToDeploy Include="..\..\javascript\DtdlErr.d.ts" />
```
[the remark this answers is not in the document]
Comment (jrdouceur) [reply samples/JsInterop/JsInterop.csproj:49]: Where do we do the copying?  Custom build step?

**Review by rido-min — APPROVED** [review]

**Comment by rido-min** [comment]
@jrdouceur make sure to squash the commits when merging

**Pull request merged**